jerry-core/parser/js/js-parser-statm.c

```diff
@@ -386,13 +394,22 @@ parser_parse_var_statement (parser_context_t *context_p) /**< context */
       parser_parse_expression (context_p,
                                PARSE_EXPR_STATEMENT | PARSE_EXPR_NO_COMMA | PARSE_EXPR_HAS_LITERAL);
     }
+#if ENABLED (JERRY_ES2015_LET_CONST_AS_VAR)
+    else
+    {
+      if (is_const)
```

Comment:
Please merge this if statement into an else if.

Reply:
Also, this check can be incorrect inside a for in/of loop. Therefore, some kind of information need to be passed to this function ( as function argument/parser context status flag) to represent whether this function has been invoked during the for in/of statement initializer part parsing or not.